### DISCUSSION TITLE
Non-assertion exceptions within AssertionScope are easy to miss

### DISCUSSION THREAD

**Discussion by aracg** [fluentassertions/fluentassertions#2411] [Q&A] [answered]
This is related to https://github.com/fluentassertions/fluentassertions/discussions/1780.

Consider this code:
```
using (new AssertionScope())
{
    1.Should().Be(2);
    throw new Exception("sample exception");
    2.Should().Be(3);
}
```
After running this I get:
```
  Message: 
Expected value to be 2, but found 1.

  Stack Trace: 
LateBoundTestFramework.Throw(String message)
TestFrameworkProvider.Throw(String message)
CollectingAssertionStrategy.ThrowIfAny(IDictionary`2 context)
Tests.Test1() line 15
```

Is this behavior expected? 
In my case, I had several calls to different methods performing specialized assertions. One of them was throwing an exception. Identifying (even noticing) the problem was difficult, it looked like the failure was only due to some assertions that failed.
Not sure how feasible is this, but from user perspective it would be very helpful if there's additional information or warning if there were other exceptions that happened within AssertionScope. 
Any thoughts/suggestions?

**Comment by dennisdoomen** [comment, marked as answer]
This is the behavior of how .NET exception handling works. The `AssertionScope` will capture all assertion failures until the assertion scope is disposed. And since you throw an exception between the two calls to `Should`, the second `Should` is never executed. Instead, the scope is disposed at the end of the curly brace, hiding the original exception. There's nothing we can do about it.
